ete4/smartview/ete/gardening.pyx

```diff
@@ -129,23 +128,25 @@ def remove(node):
         parent = parent.up
 
 
-def update_size(node):
+def update_size(node, recursive=False):
     if node.is_leaf():
         node.size = (node.dist, 1)
-    sumlengths, nleaves = get_size(node.children)
-    node.size = (abs(node.dist) + sumlengths, max(1, nleaves))
+    sumdists, nleaves = get_size(node.children, recursive)
+    node.size = (abs(node.dist) + sumdists, max(1, nleaves))
 
 
-cdef (double, double) get_size(nodes):
+cdef (double, double) get_size(nodes, recursive=False):
```

Comment:
NOT the most elegant approach passing a `recursive` parameter as argument to these two functions...

Pros
* Avoids code repetition 
* Enables efficient usage in other functions other than `standardize`
* Avoids including the function in the Tree class/newick parser (although I may prefer this option..)

Cons
* Function does not have a unique purpose
* Neither does `standardize`

Reply:
@jordibc what do you think?

Reply:
I think it's perfectly valid.

If you want another option, you could keep the old `update_size` and add something like this too:

```python
def update_all_sizes(tree):
    for node in tree.children:
        update_all_sizes(node)
    update_size(tree)
```

Reply:
Which gives you the same recursivity, avoids more or less the same code duplication, and is a bit more isolated from the previous use.

But again, both are fine. You may want to add tests to check the correctness of the new recursive functionality, by the way!